src/app/icd/ICDManager.cpp

```diff
@@ -98,11 +104,60 @@ bool ICDManager::SupportsFeature(Feature feature)
     bool success        = (Attributes::FeatureMap::Get(kRootEndpointId, &featureMap) == EMBER_ZCL_STATUS_SUCCESS);
     return success ? ((featureMap & to_underlying(feature)) != 0) : false;
 #else
-
     return ((mFeatureMap & to_underlying(feature)) != 0);
 #endif // !CONFIG_BUILD_FOR_HOST_UNIT_TEST
 }
 
+void ICDManager::SendCheckInMsgs()
+{
+#if !CONFIG_BUILD_FOR_HOST_UNIT_TEST
+    VerifyOrDie(mStorage != nullptr);
+    VerifyOrDie(mFabricTable != nullptr);
+    for (const auto & fabricInfo : *mFabricTable)
+    {
+        uint16_t supported_clients = ICDManagementServer::GetInstance().GetClientsSupportedPerFabric();
+
+        ICDMonitoringTable table(*mStorage, fabricInfo.GetFabricIndex(), supported_clients /*Table entry limit*/,
+                                 mSymmetricKeystore);
+        if (!table.IsEmpty())
+        {
+            Messaging::ExchangeManager * exchangeManager = InteractionModelEngine::GetInstance()->GetExchangeManager();
+
+            for (uint16_t i = 0; i < table.Limit(); i++)
+            {
+                ICDMonitoringEntry entry(mSymmetricKeystore);
+                if (CHIP_NO_ERROR != table.Get(i, entry))
+                {
+                    // No more entries in the table
+                    break;
+                }
```

Comment:
`table.Get(...)` can returns errors that do not indicate that the table is empty.
We need to check the specific action a take the appropriate action.

Reply:
Is you consult the implementation of the ICD Monitoring Table (I guess you've already done so since you've reviewed the PR #26809 ) This is the only way to iterate through all elements of the table. No action needs to be done upon failure of the `Get` command since what the `Get` command do is to load something from persistent storage. It's either there or not.

Reply:
ICDMonitoringTable, in the `IsEmpty` function explicitly checks that the `Get` function returns `CHIP_ERROR_NOT_FOUND` to determine if the table is empty or not. We should the same.

```
1. if table.Get == CHIP_ERROR_NOT_FOUND  -> table is empty we can break
2. if table.Get == CHIP_NO_ERROR -> Continue
3. if table.Get != CHIP_NO_ERROR && table.Get != CHIP_ERROR_NOT_FOUND -> Error occured but the table might not be empty
```
For the third case, we need to device if we check the next entry or error out of the whole process.